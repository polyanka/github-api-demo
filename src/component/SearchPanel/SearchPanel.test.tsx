import { mount, shallow } from 'enzyme';
import * as React from 'react';

import { SearchPanel } from './index';

describe('<SearchPanel /> ', () => {
  const fn = jest.fn();

  it('renders correctly with default props', () => {
    const wrapper = shallow(<SearchPanel searchValue='' setSearchValue={fn} />);
    expect(wrapper).toMatchSnapshot();
  });

  it('renders correctly if prop searchValue set login', () => {
    const wrapper = shallow(<SearchPanel searchValue='login' setSearchValue={fn} />);
    expect(wrapper).toMatchSnapshot();
  });

  it('should call onChange prop with input value', () => {
    const event = {
      preventDefault() {},
      target: { value: 'custom value' },
    };

    const wrapper = mount(<SearchPanel searchValue='' setSearchValue={fn} />);
    wrapper.find('input').simulate('change', event);
    expect(fn).toBeCalledWith('custom value');
  });
});
