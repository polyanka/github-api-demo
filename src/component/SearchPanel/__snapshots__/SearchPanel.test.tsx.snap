// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SearchPanel />  renders correctly if prop searchValue set login 1`] = `
<WithStyles(ForwardRef(Grid))
  container={true}
>
  <WithStyles(ForwardRef(FormControl))
    fullWidth={true}
    variant="outlined"
  >
    <WithStyles(ForwardRef(InputLabel))
      htmlFor="Search"
    >
      Login
    </WithStyles(ForwardRef(InputLabel))>
    <WithStyles(ForwardRef(OutlinedInput))
      labelWidth={40}
      onChange={[Function]}
      value="login"
    />
  </WithStyles(ForwardRef(FormControl))>
</WithStyles(ForwardRef(Grid))>
`;

exports[`<SearchPanel />  renders correctly with default props 1`] = `
<WithStyles(ForwardRef(Grid))
  container={true}
>
  <WithStyles(ForwardRef(FormControl))
    fullWidth={true}
    variant="outlined"
  >
    <WithStyles(ForwardRef(InputLabel))
      htmlFor="Search"
    >
      Login
    </WithStyles(ForwardRef(InputLabel))>
    <WithStyles(ForwardRef(OutlinedInput))
      labelWidth={40}
      onChange={[Function]}
      value=""
    />
  </WithStyles(ForwardRef(FormControl))>
</WithStyles(ForwardRef(Grid))>
`;
